src/rank/rank9.rs

```diff
@@ -45,6 +47,28 @@ impl Level2 {
 
         self.0 = old_part | new_part;
     }
+
+    #[cfg(target_pointer_width = "32")]
+    fn get(&self, t: usize) -> u64 {
+        debug_assert!(t < 8);
+
+        let t = t.wrapping_sub(1);
+        let shift = t.wrapping_add(t >> 28 & 8) * 9;
+        self.0 >> shift & 0x1FF
+    }
+
+    #[cfg(target_pointer_width = "32")]
+    fn set(&mut self, t: usize, value: u64) {
+        debug_assert!(t < 8);
+
+        let t = t.wrapping_sub(1);
+        let shift = t.wrapping_add(t >> 28 & 8) * 9;
+
+        let old_part = self.0 & !(0x1FF << shift);
+        let new_part = (value & 0x1FF) << shift;
+
+        self.0 = old_part | new_part;
+    }
```

Comment:
While the changes make compilation on 32-bit targets work, I'm not sure how to write tests to check if these methods are correct. Any pointers on how to get started?

Reply:
OK, so I managed to run the tests in a 32-bit platform: `wasm32-wasi` (thanks to [cargo-wasi](https://github.com/alexcrichton/cargo-wasi).

Tests are still working after `$ cargo wasi test`, which... is good? But I still don't know if the tests are hitting the corner cases for 32-bit pointers.

Reply:
Upon further reading, these changes still work fine, but might not be the fastest (because `u64` will need to be emulated by 32-bit targets, just like `u128` would need to be emulated in 64-bit targets). Should I leave a note for the future?